keras/backend/theano_backend.py

```diff
@@ -1686,3 +1692,29 @@ def ctc_step(y_true_step, y_pred_step, input_length_step, label_length_step):
 
     ret = ret.dimshuffle('x', 0)
     return ret
+
+
+class ReverseGradient(theano.Op):
+    '''Flips the sign of incoming gradient during training.'''
+    view_map = {0: [0]}
```

Comment:
This probably doesn't do what you think it does. Do not use global class attributes, especially not ones that are pointers.